main/openrc/firstboot.initd

```diff
@@ -0,0 +1,44 @@
+#!/sbin/openrc-run
+
+# The first boot init service
+
+# read kernel options
+init_KOPT() {
+	for opt in $(cat /proc/cmdline 2>/dev/null); do
+		case "$opt" in
+			ssh_*=*)
+				eval "KOPT_${opt%%=*}='${opt#*=}'" ;;
+		esac
+	done
+}
+
+depend() {
+	keyword -vserver -lxc
+	init_KOPT
+	if [ -n "$KOPT_ssh_key" ] || [ -n "$KOPT_ssh_pass" ]; then
+		need sshd
+	fi
+}
+
+
+start() {
+	init_KOPT
+	local rc=0
+	ebegin "Starting ${RC_SVCNAME}"
+	if [ -n "$KOPT_ssh_key" ] && [ ! -f "/root/.ssh/authorized_keys" ]; then
+		einfo "Fetching ssh keys"
+		mkdir -p /root/.ssh
+		case "$KOPT_ssh_key" in
+			http://*|https://*|ftp://*|ftps://*)
+				wget -q "$KOPT_ssh_key" -O /root/.ssh/authorized_keys
+				rc=$?;;
+			*) echo "$KOPT_ssh_key" > /root/.ssh/authorized_keys;;
+		esac
+	fi
+	if [ -n "$KOPT_ssh_pass" ]; then
+		echo "root:$KOPT_ssh_pass" | /usr/sbin/chpasswd
+		SSHD_OPTS="-o PermitRootLogin=yes" rc-service sshd start
```

Comment:
Are you sure you can pass env vars to rc-service? I would presume not.

I think an alternative approach would be to add this option to openssh initd

```
--- sshd
+++ sshd.new
@@ -4,7 +4,7 @@
 description_checkconfig="Verify configuration file"
 description_reload="Reload configuration"
 
-extra_commands="checkconfig"
+extra_commands="checkconfig start_permit_root"
 extra_started_commands="reload"
 
 # NOTE: SSHD_* variables are deprecated and will be removed in future!
@@ -97,6 +97,11 @@
 	start-stop-daemon --signal HUP \
 		--exec "$command" --pidfile "$pidfile"
 	eend $?
+}
+
+start_permit_root() {
+	command_args="$command_args -o PermitRootLogin=yes"
+	start
 }
 
 warn_deprecated_var() {
```

Reply:
Updated. Thanks.

Reply:
I tried this and `start_permit_root()` does not play nice with `stop/start/restart`. I have to stick with current approach.